assets/model_monitoring/components/src/generation_safety_quality/annotation_compute_histogram/run.py

```diff
@@ -1558,10 +1559,21 @@ def apply_annotation(
         SIMILARITY: [prompt_column_name, completion_column_name, ground_truth_column_name]
     }
     # Sampling
-    production_df = production_df.sample(withReplacement=False, fraction=sample_rate)
-    production_df_with_index = production_df.withColumn("id",
-                                                        row_number()
-                                                        .over(Window.orderBy(monotonically_increasing_id()))-1)
+    production_df_sampled = production_df.sample(withReplacement=False, fraction=sample_rate)
+    if production_df_sampled.count() == 0:
+        print("Not enough data resulting from sample_rate and production dataset. "
```

Comment:
We seem to be re-choosing the sample_rate for the user. Should you inform them about that in the print? The print statement looks like it is expecting user to fix their sample_rate parameter. Should this be an exception.

The steps taken from lines 1563 to 1570 seem a little confusing to me.

Reply:
updated a little. we discussed this and don't want to fail the job in this case so don't want to throw an exception but inform the user that we will not honor their sample_rate